### DISCUSSION TITLE
option to copy joystick Up onto Sega A or C

### DISCUSSION THREAD

**Discussion by W-Livi** [UzixLS/zx-sizif-512#29] [Ideas]
It is tremendously cool that Sizif 512 supports a six-button Sega gamepad. I especially appreciate the convenience of using Mode to open the magic menu, so it's easy to switch between Kempston and Sinclair as required.  However, it feels like the gamepad's A and C buttons are a little underutilized. It's great to have buttons 2 and 3 where supported (like using Kempston to navigate the esxdos long filename browser), but it would be nice if the extra buttons could have something to do while running software that ignores them.

Suggestion: a configuration flag (controlled from the magic menu) to turn the A or C button into a second Joystick Up input.
why Up? because many platform games used Up to jump; a control which would usually get a button all its own on later systems.

I'm not about to suggest an entire controller remapping system, as that's a lot of settings, and would doubtless require a lot of CPLD changes to make it work.  I'm hoping that requesting a single flag to remap a single button will be a smaller, somewhat more manageable request.

**Comment by UzixLS** [comment]
Many games provide option to redefine keys, you can switch joystick to Sinclair mode (in which joystick actually emulates keyboard) and map jump to any joystick key.

**Comment by W-Livi** [comment]
Fair enough; I guess I was kinda overlooking the Sinclair mode when Kempston was available

**Comment by UzixLS** [comment]
Despite my previous message, I've found that having UP on two keys (Up and A) simultaneously in some games is more convenient than just remap. I've implemented this option in 36d77c3ab6b3c0578d361faf107460fa58fc1c7e
